scripts/west_commands/runners/qemu.py

```diff
@@ -4,28 +4,210 @@
-        # This is a stub.
-        return RunnerCaps(commands=set())
+        return RunnerCaps(commands={'flash', 'debug', 'debugserver', 'attach'},
+                          tool_opt=True, file=True)
 
     @classmethod
     def do_add_parser(cls, parser):
-        pass                    # Nothing to do.
+        # Required:
+        parser.add_argument('--machine', required=True, help='machine name')
+
+        # Optional:
+        parser.add_argument('--cpu', required=False, dest='cpu',
+                            help='specifies a cpu type')
+        parser.add_argument('--tui', default=False, action='store_true',
+                            help='if given, GDB uses -tui')
+        parser.add_argument('--gdbserver', default='JLinkGDBServer',
+                            help='GDB server, default is JLinkGDBServer')
+        parser.add_argument('--gdb-host', default='',
+                            help='custom gdb host, defaults to the empty string '
+                            'and runs a gdb server')
+        parser.add_argument('--gdb-port', default=DEFAULT_QEMU_GDB_PORT,
+                            help='pyocd gdb port, defaults to {}'.format(
+                                DEFAULT_QEMU_GDB_PORT))
+        parser.add_argument('--dtb', required=False, dest='dtb',
+                            help='specifies a dtb file')
+        parser.add_argument('--qemu_option', required=False, dest='qemu_opts',
+                            help='specifies qemu opts')
+        parser.add_argument('--commander', default=DEFAULT_QEMU_EXE,
+                            help=f'''QEMU executable, default is
+                            {DEFAULT_QEMU_EXE}''')
+        parser.add_argument('--pipe', default=DEFAULT_QEMU_FIFO,
+                            help=f'''if given, qemu use pipe''')
 
     @classmethod
     def do_create(cls, cfg, args):
-        return QemuBinaryRunner(cfg)
+        return QemuBinaryRunner(cfg, machine=args.machine,
+                                 cpu=args.cpu,
+                                 dtb=args.dtb,
+                                 qemu_opts=args.qemu_opts,
+                                 commander=args.commander,
+                                 pipe=args.pipe,
+                                 gdbserver=args.gdbserver,
+                                 gdb_host=args.gdb_host,
+                                 gdb_port=args.gdb_port,
+                                 tui=args.tui, tool_opt=args.tool_opt)
 
     def do_run(self, command, **kwargs):
-        pass
+        self.commander = os.fspath(
+            Path(self.require(self.commander)).resolve())
+
+        #server_cmd = ([self.gdbserver] +
+        #              ['-port', str(self.gdb_port)] + self.tool_opt)
+
+        if command == 'flash':
+            self.flash(**kwargs)
+        elif command == 'debug':
+            self.debug(**kwargs)
+        else:
+            self.logger.warnning("%s not supported yet", format(command))
```

Comment:
Please remove this path entirely. This is what `RunnerCaps(command=...)` is for.

Reply:
thanks,  I need updat to the capabilities routine.